apps/gallery/js/thumbnail_list.js

```diff
@@ -35,9 +35,9 @@ function ThumbnailList(groupClass, container) {
   this.groupClass = groupClass;
   this.container = container;
 
-  // If the locale changes while we're running, then we need to
-  // re-localize the dates in the group headers
-  navigator.mozL10n.ready(this.localize.bind(this));
+  // Localize dates in group headers and thumbnail item
+  // description when l10n resources are loaded
+  navigator.mozL10n.once(this.localize.bind(this));
```

Comment:
You are right, using  once l10n handler inside thumbnailList constructor and waiting for all thumbnails to be created before setting ready call in startup.js is causing localized call twice on app start. 

I will move the call to thumbnails.js right after we are creating instance of thumbnailList.